src/login.c

```diff
@@ -1111,7 +1111,7 @@ int main (int argc, char **argv)
 #endif
 
 	/* If we were init, we need to start a new session */
-	if (getppid() == 1) {
+	if (1 == initial_pid) {
```

Comment:
Please mention which commit(s) this is fixing.  See 'Fixes:' tags in existing commits.